src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+
+			const valueId: ValueID = {
+				commandClass: CommandClasses["Color Switch"],
+				endpoint: endpoint.index,
+				property: "supportedColor",
+				propertyKey: key,
+			};
+
+			// TODO: Should we be setting this every run?
+			node.valueDB.setMetadata(valueId, {
+				...ValueMetadata.ReadOnly,
+				label: `Supports ${key}`,
+				description: `Whether the endpoint supports setting the ${key} color.`,
+			});
+			node.valueDB.setValue(
+				valueId,
+				this._supportedColors[supportsColorKey] ?? false,
+			);
+		}
+	}
+
+	private _supportedColors: SupportedColorTable | null;
+
+	public get supportsWarmWhite(): boolean | undefined {
+		return this._supportedColors?.supportsWarmWhite;
+	}
+
+	public get supportsColdWhite(): boolean | undefined {
+		return this._supportedColors?.supportsColdWhite;
+	}
+
+	public get supportsRed(): boolean | undefined {
+		return this._supportedColors?.supportsRed;
+	}
+
+	public get supportsGreen(): boolean | undefined {
+		return this._supportedColors?.supportsGreen;
+	}
+
+	public get supportsBlue(): boolean | undefined {
+		return this._supportedColors?.supportsBlue;
+	}
+
+	public get supportsAmber(): boolean | undefined {
+		return this._supportedColors?.supportsAmber;
+	}
+
+	public get supportsCyan(): boolean | undefined {
+		return this._supportedColors?.supportsCyan;
+	}
+
+	public get supportsPurple(): boolean | undefined {
+		return this._supportedColors?.supportsPurple;
+	}
+
+	public get supportsIndex(): boolean | undefined {
+		return this._supportedColors?.supportsIndex;
+	}
+
+	public toJSON(): JSONObject {
+		return super.toJSONInherited({
+			...this._supportedColors,
+		});
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedGet)
+@expectedCCResponse(ColorSwitchCCSupportedReport)
+export class ColorSwitchCCSupportedGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | CCCommandOptions,
+	) {
+		super(driver, options);
+	}
+}
+
+@CCCommand(ColorSwitchCommand.Report)
+export class ColorSwitchCCReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		if (this.version >= 3) {
+			validatePayload(this.payload.length >= 4);
+		} else {
+			validatePayload(this.payload.length >= 2);
+		}
+		this._colorComponent = this.payload[0];
+		this._currentValue = this.payload[1];
+
+		if (this.version >= 3) {
+			this._targetValue = this.payload[2];
+			this._duration = Duration.parseReport(this.payload[3]);
+		}
+
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const colorName = getColorKeyFromComponent(this._colorComponent);
+		if (!colorName) {
+			return;
+		}
+
+		const valueId: ValueID = {
+			commandClass: CommandClasses["Color Switch"],
+			property: "currentColor",
+			endpoint: endpoint.index,
+			propertyKey: colorName,
+		};
+
+		// TODO: Should we be setting this every run?
```

Comment:
You should probably create all the metadata in the interview procedure. See Notification CC line 284ff (and I'm sure I've done this somewhere else too).